src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java

```diff
@@ -1435,6 +1431,25 @@ public Object visit(BBOX filter, Object extraData) {
         return dataQuery;
     }
 
+    private CoordinateReferenceSystem getDeclaredCRS(
+            FeatureSource<? extends FeatureType, ? extends Feature> source,
+            String wfsVersion,
+            CoordinateReferenceSystem crs,
+            final FeatureTypeInfo featureTypeInfo) {
+        if (source.getDataStore() instanceof DataStore) {
+            return WFSReprojectionUtil.getDeclaredCrs(crs, wfsVersion);
+        } else {
+            final ProjectionPolicy projectionPolicy = featureTypeInfo.getProjectionPolicy();
```

Comment:
This logic seems like it could be shared with the simple feature case really?

Reply:
The condtion separates these cases:
- implements DataStore interface? -> It is a simple-features store, so return normal declared CRS (refactored only, no changes for this case).
- does not implement DataStore interface? -> it should be a complex features store, so execute new-fixed logic.

Since in WFS GetFeature we don't have access to AppSchemaDataAccess class I used the DataStore interface as the differentiator between SimpleFeature and ComplexFeature Datastore, so now the question is: is this the right way to separate these cases?

As note I see other no DataStore subclasses are DecoratingDataAcess, ReadOnlyDataAccess and WFSContentDataAccess.

There is a safer way to check it is a complex features case?  Wondering if checking the feature type implements SimpleFeatureType or not is a safer place?

Reply:
Changes done, now we are checking the involved feature type instead the datastore interface.